File: src/main/java/com/projectswg/holocore/resources/support/global/commands/callbacks/chat/CmdSpatialChatInternal.java
package com.projectswg.holocore.resources.support.global.commands.callbacks.chat;

import com.projectswg.holocore.intents.support.global.chat.SpatialChatIntent;
import com.projectswg.holocore.resources.support.global.commands.ICmdCallback;
import com.projectswg.holocore.resources.support.global.player.Player;
import com.projectswg.holocore.resources.support.objects.swg.SWGObject;
import org.jetbrains.annotations.NotNull;

public final class CmdSpatialChatInternal implements ICmdCallback {
	
	@Override
	public void execute(@NotNull Player player, SWGObject target, @NotNull String args) {
		String[] cmdArgs = args.split(" ", 6);
		
		new SpatialChatIntent(player, Integer.parseInt(cmdArgs[1]), args.substring(10), Integer.parseInt(cmdArgs[2]), Integer.parseInt(cmdArgs[4])).broadcast();
	}
	
}

File: src/main/java/com/projectswg/holocore/resources/support/global/commands/callbacks/chat/CmdSpatialChatInternal.kt
package com.projectswg.holocore.resources.support.global.commands.callbacks.chat

import com.projectswg.holocore.intents.support.global.chat.SpatialChatIntent
import com.projectswg.holocore.resources.support.global.commands.ICmdCallback
import com.projectswg.holocore.resources.support.global.player.Player
import com.projectswg.holocore.resources.support.objects.swg.SWGObject

class CmdSpatialChatInternal : ICmdCallback {
	override fun execute(player: Player, target: SWGObject?, args: String) {
		val cmdArgs = args.split(" ".toRegex(), limit = 6).toTypedArray()

		SpatialChatIntent(player, cmdArgs[1].toInt(), args.substring(10), cmdArgs[2].toInt(), cmdArgs[4].toInt()).broadcast()
	}
}
